src/BingAIClient.js

```diff
@@ -468,10 +513,36 @@ export default class BingAIClient {
                             // delete useless suggestions from moderation filter
                             delete eventMessage.suggestedResponses;
                         }
-                        resolve({
-                            message: eventMessage,
-                            conversationExpiryTime: event?.item?.conversationExpiryTime,
-                        });
+                        if (bicIframe) {
+                            // the last messages will be a image creation event if bicIframe is present.
+                            let i = messages.length - 1;
+                            while (eventMessage?.contentType === 'IMAGE' && i > 0) {
+                                eventMessage = messages[i -= 1];
+                            }
+
+                            // wait for bicIframe to be completed.
+                            bicIframe.then(async (result) => {
+                                // The frame can be large, only put it into adaptiveCards.
+                                eventMessage.adaptiveCards[0].body[0].text += result;
+                                resolve({
+                                    message: eventMessage,
+                                    conversationExpiryTime: event?.item?.conversationExpiryTime,
+                                });
+                            }).catch((error) => {
+                                eventMessage.text += `<br>${error}`;
+                                eventMessage.adaptiveCards[0].body[0].text = eventMessage.text;
+                                resolve({
+                                    message: eventMessage,
+                                    conversationExpiryTime: event?.item?.conversationExpiryTime,
+                                });
```

Comment:
I don't think `async`  is needed in the `then()`.

Additionally, I would prefer you use the try-catch-await pattern instead.

Reply:
Yes, it's not needed. I tried some different processing methods here before and forgot to remove 'async'.
I have tried to use the try-catch-await pattern at first, but it needs to modify 'ws.on('message', (data) => ' to 'ws.on('message', async (data) => '.
I'm not sure if it's a good idea to do this.

Reply:
Go ahead and make that change. I think the code will be cleaner that way. Plus maybe it will fix the other issue with the error causing the entire API server to exit.

Reply:
OK. I have changed it to the try-catch-await pattern. But I'm not sure if it can fix the issue which causes the entire API server to exit with error.